test/test_mps.py

```diff
@@ -61,6 +61,10 @@
     )
 )
 
+def xfailIfMacOS14_4(func):
+    return unittest.expectedFailure(func) if product_version > 14.3 else func  # noqa: F821
```

Comment:
This will be confusing when macos-m1-14 updates to 14.5.
Maybe `if 14.3 < product_version < 14.5`?

Reply:
No, assumption is that thins are broken until fixed. But good point, let me rename it `MacOS14_4Plus`